File: app/src/main/java/com/example/demo/sundu/developer/netingscrollview/two/NestingScrollActivity.java
package com.example.demo.sundu.developer.netingscrollview.two;

import android.os.Bundle;
import android.util.SparseArray;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.example.demo.R;
import com.example.demo.sundu.developer.netingscrollview.NestingScrollRecyclerAdapter;
import com.google.android.material.tabs.TabLayout;

;

/**
 * @author cginechen
 * @date 2016-12-28
 */

public class NestingScrollActivity extends AppCompatActivity {
    private Toolbar mToolbar;
    private TabLayout mTabLayout;
    private ViewPager mViewPager;
    private SparseArray<RecyclerView> mPageMap = new SparseArray<>();
    private PagerAdapter mPagerAdapter = new PagerAdapter() {
        @Override
        public boolean isViewFromObject(View view, Object object) {
            return view == object;
        }

        @Override
        public int getCount() {
            return 3;
        }

        @Override
        public Object instantiateItem(final ViewGroup container, int position) {
            View view = getPageView(position);
            ViewGroup.LayoutParams params = new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
            container.addView(view, params);
            return view;
        }

        @Override
        public void destroyItem(ViewGroup container, int position, Object object) {
            container.removeView((View) object);
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return "item " + (position + 1);
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_nestiong_scroll_plan_nesting_scroll);
        mToolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(mToolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        mTabLayout = (TabLayout) findViewById(R.id.tab_layout);
        mViewPager = (ViewPager) findViewById(R.id.viewpager);
        mViewPager.setAdapter(mPagerAdapter);
        mTabLayout.setupWithViewPager(mViewPager);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            overridePendingTransition(R.anim.slide_still, R.anim.slide_out_right);
        }

        return super.onOptionsItemSelected(item);
    }

    private RecyclerView getPageView(int pos) {
        RecyclerView view = mPageMap.get(pos);
        if (view == null) {
            RecyclerView recyclerView = new RecyclerView(this);
            recyclerView.setLayoutManager(new LinearLayoutManager(this));
            recyclerView.setAdapter(new NestingScrollRecyclerAdapter());
            mPageMap.put(pos, recyclerView);
            return recyclerView;
        }
        return view;
    }

}

File: app/src/main/java/com/example/demo/sundu/developer/netingscrollview/NestingScrollActivity.kt
package com.example.demo.sundu.developer.netingscrollview

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.example.demo.R
import com.example.demo.sundu.developer.netingscrollview.one.EventDispatchPlanActivity
import com.example.demo.sundu.developer.netingscrollview.three.CoordinatorLayoutActivity
import com.example.demo.sundu.developer.netingscrollview.two.NestingScrollActivity
import java.util.*

class NestingScrollActivity : AppCompatActivity() {
    private var mToolbar: Toolbar? = null
    private var mListView: ListView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_nesting_scrollerview_layout)
        mToolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(mToolbar)
        val mData = ArrayList<String>()
        mData.add("原始事件分发方案")
        mData.add("使用NestScrollParent与NestScrollChild接口")
        mData.add("使用CoordinatorLayout")
        mListView = findViewById<View>(R.id.listView) as ListView
        mListView!!.adapter =
            ArrayAdapter(this, android.R.layout.simple_list_item_1, mData)
        mListView!!.onItemClickListener =
            AdapterView.OnItemClickListener { parent, view, position, id ->
                var intent: Intent? = null
                if (position == 0) {
                    intent =
                        Intent(this@NestingScrollActivity, EventDispatchPlanActivity::class.java)
                } else if (position == 1) {
                    intent = Intent(this@NestingScrollActivity, NestingScrollActivity::class.java)
                } else if (position == 2) {
                    intent =
                        Intent(this@NestingScrollActivity, CoordinatorLayoutActivity::class.java)
                }
                if (intent != null) {
                    startActivity(intent)
                    overridePendingTransition(R.anim.slide_in_right, R.anim.slide_still)
                }
            }
    }
}
